content-api/content-actors/src/main/scala/org/sunbird/content/util/RetireManager.scala

```diff
@@ -0,0 +1,120 @@
+package org.sunbird.content.util
+
+import java.util
+import java.util.{Date, UUID}
+
+import org.apache.commons.collections4.CollectionUtils
+import org.apache.commons.lang.StringUtils
+import org.sunbird.cache.impl.RedisCache
+import org.sunbird.common.{DateUtils, JsonUtils, Platform}
+import org.sunbird.common.dto.{Request, Response, ResponseHandler}
+import org.sunbird.common.exception.{ClientException, ResourceNotFoundException, ServerException}
+import org.sunbird.graph.dac.model.Node
+import org.sunbird.graph.OntologyEngineContext
+import org.sunbird.graph.external.ExternalPropsManager
+import org.sunbird.graph.nodes.DataNode
+import org.sunbird.graph.utils.ScalaJsonUtils
+import org.sunbird.kafka.client.KafkaClient
+import org.sunbird.parseq.Task
+import org.sunbird.telemetry.logger.TelemetryManager
+import org.sunbird.utils.HierarchyConstants
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.{ExecutionContext, Future}
+
+object RetireManager {
+    val finalStatus: util.List[String] = util.Arrays.asList("Flagged", "Live", "Unlisted")
+    private val kfClient = new KafkaClient
+
+    def retire(request: Request)(implicit ec: ExecutionContext, oec: OntologyEngineContext): Future[Response] = {
+        validateRequest(request)
+        getNodeToRetire(request).flatMap(node => {
+            val futureList = Task.parallel[Response](
+                handleCollectionToRetire(node, request),
```

Comment:
Based on mimeType call two methods which will internally handle neo4j and cassandra update together.